sdk/resourcemanager/azure-resourcemanager-network/src/main/java/com/azure/resourcemanager/network/implementation/NetworkInterfaceImpl.java

```diff
@@ -563,9 +579,41 @@ protected void beforeCreating() {
                 .withNetworkSecurityGroup(new NetworkSecurityGroupInner().withId(networkSecurityGroup.id()));
         }
 
-        NicIpConfigurationImpl.ensureConfigurations(this.nicIPConfigurations.values());
+        NicIpConfigurationImpl.ensureConfigurations(this.nicIPConfigurations.values(), this.specifiedIpConfigNames);
 
         // Reset and update IP configs
         this.innerModel().withIpConfigurations(innersFromWrappers(this.nicIPConfigurations.values()));
     }
+
+    @Override
+    public NetworkInterfaceImpl withPrimaryPublicIPAddressDeleteOptions(DeleteOptions deleteOptions) {
+        this.ensureDeleteOptions(deleteOptions, "primary");
+        return this;
+    }
+
+    @Override
+    public NetworkInterfaceImpl update() {
+        this.specifiedIpConfigNames = new HashMap<DeleteOptions, HashSet<String>>();
+        this.isInUpdateIpConfigMode = false;
+        return super.update();
+    }
+
+    @Override
+    public NetworkInterface apply() {
+        if (!isInCreateMode() && !isInUpdateIpConfigMode && !this.specifiedIpConfigNames.isEmpty()) {
+            this.specifiedIpConfigNames.forEach((deleteOptions, values) -> this.nicIPConfigurations.values()
+                    .stream().filter(ipConfig -> values.contains(ipConfig.name())
+                            && Objects.nonNull(ipConfig.innerModel().publicIpAddress()))
+                    .forEach(ipConfig -> ipConfig.innerModel().publicIpAddress().withDeleteOption(deleteOptions)));
+        }
```

Comment:
I don't quite get this part. 
1. Why this "set PublicIp deleteOptions" logic in `apply()` again, when it's already in `beforeCreate`? I assume `beforeCreate` will also be called in update flow. 
2. Can we make the `specifiedIpConfigNames` Map<String, DeleteOptions>`, with key of ipConfigName? Or are there any concerns not to do that?
3. If above is true, then `isInUpdateIpConfigMode` seems no longer needed, right? We can always set deleteOptions when the corresponding value is not null.

Reply:
Your idea is working after live test and has been fixed in the new version.